docs/src/pages/components/buttons/CustomVariantButtons.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {
+  createMuiTheme,
+  makeStyles,
+  ThemeProvider,
+} from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+
+const useStyles = makeStyles((theme) => ({
+  margin: {
+    margin: theme.spacing(1),
+  },
+}));
+
+const theme = createMuiTheme({
+  variants: {
+    MuiButton: [
+      {
+        matcher: { variant: 'dashed' },
+        styles: {
```

Comment:
Should we specify root here as well as other nested component selectors here?

Reply:
The support of multiple style rules will add complexity. The current alternative to customize nested element is with the global CSS API, e.g. `& .MuiButton-label`. The main downside is to increase specificity, making it hard for people using the design to customize it. I'm not against:

```suggestion
        styles: {
          root: {
            padding: '5px 15px',
            border: '5px dashed grey',
          },
          label: {
            color: 'blue';
          },
        },
```

But I'm not sure it worth it.

cc @mbrookes is this something you would have needed in your iOS theme experimentation?